@here/harp-mapview/lib/DecodedTileHelpers.ts

```diff
@@ -71,7 +71,7 @@ export interface MaterialOptions {
  */
 export function createMaterial(
     options: MaterialOptions,
-    materialUpdateCallback?: () => void
+    materialUpdateCallback?: (texture: THREE.Texture) => void
```

Comment:
Not sure why this is needed.
BTW. both landmark and should call this `materialUpdateCallback` right after their texture is ready, but ... they shoula also call `<TIle>this.addOwnedTexture(t)`, so i don't see reason to leak this texture in this particular place.

Btw, `WebTileDataSource` should be updated to remove "my" WebTile and just call `addOwnedTexture`.

Reply:
> Not sure why this is needed.

Ok, i get it, it's needed because `createMaterial` creates textures.
However, this `materialUpdateCallback` is more like `textureCreatedCallback` ...

Reply:
The problem is, that createMaterial does not know anything about Tile, so I can only add the ownership of the texture to the tile in the callback and I have to do the same for the LandmarkTile (and WebTile).
The only way I see to get around this would be to make createMaterial a method of Tile but that would mean moving almost the whole DecodedTileHelpers code into Tile.

Reply:
first ;) 
Thanks i've already found explanation in companion patch.

However, `WebTileDataSource` should be updated to use `addOwnedTexture`.